metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -256,24 +247,34 @@ def _build_derived_metric_output_node(
                 )
             )
 
-        if len(parent_nodes) == 1:
-            return ComputeMetricsNode(
-                parent_node=parent_nodes[0],
-                metric_specs=[metric_spec],
-            )
-
-        return ComputeMetricsNode(
-            parent_node=CombineMetricsNode(parent_nodes=parent_nodes),
+        compute_metrics_node = ComputeMetricsNode(
+            parent_node=parent_nodes[0] if len(parent_nodes) == 1 else CombineMetricsNode(parent_nodes=parent_nodes),
             metric_specs=[metric_spec],
         )
 
+        # For nested ratio / derived metrics with time offset, apply offset after metric computation.
+        join_to_time_spine_node: Optional[JoinToTimeSpineNode] = None
```

Comment:
> @tlento: What happens if there's a filter on metric_time?
> 
> Today we generally do:
> 
> join
> filter
> aggregate
> All of this is in ComputeMetricsNode, which comes post-aggregation, so I think we'll end up joining the time spine against the filtered input set here, effectively truncating the input data on the second offset.
> 
> If I'm right about this, we'll need to propagate the offset information so we can expand that window. For now it might be adequate to apply the filter expression after the final calculation if there's a nested derived offset, and then we can optimize the predicate handling later.

Reply:
@tlento I think you're right - I'll add a test case for that.

Reply:
@tlento - updated to handle that case!